src/AssembleScalarNormalGradientBCSolverAlgorithm.C

```diff
@@ -0,0 +1,218 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+
+// nalu
+#include <AssembleScalarNormalGradientBCSolverAlgorithm.h>
+#include <EquationSystem.h>
+#include <FieldTypeDef.h>
+#include <LinearSystem.h>
+#include <Realm.h>
+#include <TimeIntegrator.h>
+#include <master_element/MasterElement.h>
+
+// stk_mesh/base/fem
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/Field.hpp>
+#include <stk_mesh/base/GetBuckets.hpp>
+#include <stk_mesh/base/GetEntities.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+#include <stk_mesh/base/Part.hpp>
+
+namespace sierra{
+namespace nalu{
+
+//==========================================================================
+// Class Definition
+//==========================================================================
+// AssembleScalarNormalGradientBCSolverAlgorithm - 
+//   This boundary conditions applies a scalar normal flux bc, 
+//   Int bcScalarQ*area, such that a specified normal temperature gradient
+//   is achieved.
+//==========================================================================
+//--------------------------------------------------------------------------
+//-------- constructor -----------------------------------------------------
+//--------------------------------------------------------------------------
+AssembleScalarNormalGradientBCSolverAlgorithm::AssembleScalarNormalGradientBCSolverAlgorithm(
```

Comment:
At first, when I thought that the implementation was going to be a flux-based approach, I felt that a general name would suffice. However, after our first meeting, it was very clear that this class can only be used for a specific equation system, e.g., enthalpy. This is due to the specific heat scaling. As such, I would change the name to tie in the EnthalpyEQS somehow. Again, a very minor detail.

Reply:
I second this. This algorithm is specific to EnthalpyEQS.